lib/Graph/Graph.cpp

```diff
@@ -977,7 +977,13 @@ CmpLTENode *Function::createCmpLTE(llvm::StringRef name, NodeValue LHS,
 CmpEQNode *Function::createCmpEQ(llvm::StringRef name, NodeValue LHS,
                                  NodeValue RHS) {
   assert(LHS.dims() == RHS.dims() && "Invalid operand shapes");
-  auto OT = getParent()->uniqueType(*LHS.getType());
+  TypeRef OT;
```

Comment:
could you extract the piece of code for getting OT type into a separate static method? (reuse that for all boolean ops).

Reply:
Done.